src/eradiate/scenes/surface/_dem.py

```diff
@@ -1,50 +1,320 @@
+def mesh_from_dem(
+    da: xr.DataArray,
+    geometry: SceneGeometry,
+    planet_radius: pint.Quantity | float = EARTH_RADIUS,
+):
     """
-    DEM surface [``dem``]
+    Construct a DEM surface from a dataarray holding elevation data.
+    There are two ways of specifying the dataarray:
+    - With latitude and longitude based coordinates:
+      The coordinates must be named "lat" and "lon" and be in that order.
+    - With x and y length based coordinates
+      The coordinates must be named "x" and "y" and be in that order.
+
+    The unit of the coordinates must be given by the `coordinate_units` attribute in the DataArray,
+    while the unit of the elevation data must be given by the `unit` attribute for automatic conversion.
 
-    A surface description for digital elevation models.
+    Parameters
+    ----------
+    da : xr.DataArray
+        DataArray with elevation data, indexed either by latitude and longitude coordinates or x and y lenghts
 
-    This surface supports instantiation based on an xarray data array, a
-    triangulated mesh file in the PLY and OBJ formats and an analytical
-    elevation mapping the x and y coordinates to an elevation value.
+    geometry : SceneGeometry
+        Geometry of the scene.
 
-    .. admonition:: Class method constructors
+    planet_radius : pint.Quantity
+        Planet radius, used for converting between latitude/longitude and x/y coordinates *ONLY* in the case
+        of a plane parallel scene geometry.
 
-       .. autosummary::
+    id : str
+        Identifier of the scene element.
 
-          from_dataarray
-          from_analytical
+    Returns
+    -------
 
-    Notes
-    -----
-    * Contrary to most other surfaces, this scene element expands as a single
-      Mitsuba Shape plugin which includes its child BSDF rather than referencing
-      a top-level dictionary entry. The reason for this is that some allowed
-      shapes expand as Mitsuba instances rather than dictionaries, which
-      prevents object referencing.
+    mesh : :class:`.BufferMeshShape`
+        A triangulated mesh representing the DEM
+
+    theta_lim : tuple
+        The limits of the latitude extent of the DEM
+
+    phi_lim : tuple
+        The limits of the longitude extent of the DEM
+    """
+
+    coord_units = da.attrs["coordinate_units"]
+    if "lat" in da.coords and "lon" in da.coords:
+        lat = ureg.Quantity(da.lat.values, coord_units).m_as(ureg.deg)
+        lon = ureg.Quantity(da.lon.values, coord_units).m_as(ureg.deg)
+        vertices = _generate_dem_vertices(lat, lon, to_quantity(da))
+        faces = _generate_face_indices(len(lat), len(lon))
+
+    elif "x" in da.coords and "y" in da.coords:
+        if isinstance(geometry, SphericalShellGeometry):
+            planet_radius = geometry.planet_radius
+
+        x = np.rad2deg(ureg.Quantity(da.x.values, coord_units) / planet_radius).m
+        y = np.rad2deg(ureg.Quantity(da.y.values, coord_units) / planet_radius).m
```

Comment:
That's odd, why not use `m_as(ureg.deg)`?

Reply:
Because the result of the division does not have `radians` as a unit. After dividing the two quantities, the result will be `unitless`.